### PR TITLE
Impl/create deb for raspberry

### PR DESCRIPTION
**Pull request by alexkutsan**
Fixes FORDTCN-5847

:a:  THIS PR WILL NOT BE MERGED
:warning:  THIS PR WILL BE USED TO PORT SDL ON RASPBERRY. IT SONTAINS CODE THAT SHOULD NOT BE IN DEVELOP BRANCH 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Tested manually basic scenarios

### Summary
Fixing the install path of SDL to make installation to the system.
Add creating deb package.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -242,11 +242,12 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
...
 # Please do not change compiler/linker flags if You do not know how particular
 # flag is handled by CMake
-set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_BINARY_DIR})
-set(ARCHIVE_OUTPUT_DIRECTORY ./bin)
+#set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_BINARY_DIR})
+set(INSTALL_LIB_DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
+set(INSTALL_BIN_DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 
 set(EXCLUDE_ERROR_FLAGS "-Wno-error=unused-variable -Wno-error=unused-but-set-variable -Wno-error=deprecated-declarations")
-set(CMAKE_CXX_FLAGS "-fPIC -std=gnu++0x -Wall -fvisibility=hidden -Werror ${EXCLUDE_ERROR_FLAGS} -Wuninitialized -Wvla")
```
Comment (AKalinich-Luxoft) [inline CMakeLists.txt:249]: @alexkutsan is it not necessary to handle all warnings for raspberry?

File: src/components/application_manager/src/plugin_manager/rpc_plugin_manager_impl.cc [hunk 2]
```diff
@@ -37,7 +37,7 @@ T GetFuncFromLib(void* dl_handle, const std::string& function_name) {
       reinterpret_cast<T>(dlsym(dl_handle, function_name.c_str()));
   char* error_string = dlerror();
   if (nullptr != error_string) {
-    LOG4CXX_ERROR(logger_, "Failed to export symbols : " << error_string);
+    LOG4CXX_DEBUG(logger_, "Failed to export symbols : " << error_string);
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/plugin_manager/rpc_plugin_manager_impl.cc:40]: @alexkutsan why is this not considered as error?

File: src/components/security_manager/src/ssl_context_impl.cc [hunk 3]
```diff
@@ -596,7 +596,7 @@ void CryptoManagerImpl::SSLContextImpl::ResetConnection() {
     SSL_shutdown(connection_);
   }
   LOG4CXX_DEBUG(logger_, "SSL connection recreation");
-  SSL_CTX* ssl_context = connection_->ctx;
+  SSL_CTX* ssl_context = SSL_get_SSL_CTX(connection_);
```
Comment (AKalinich-Luxoft) [inline src/components/security_manager/src/ssl_context_impl.cc:599]: @alexkutsan is this change necessary?

File: src/appMain/smartDeviceLink.ini [hunk 4]
```diff
@@ -37,10 +37,10 @@ LogsEnabled = true
 AppConfigFolder =
 ; Contains SDL configuration files - .json/.ini
 ; Default value is SDL working directory
-AppStorageFolder = storage
+AppStorageFolder = /tmp/sdl/storage
 ; Contains resourses, e.g. audio8bit.wav
 ; Default value is SDL working directory
-AppResourceFolder =
+AppResourceFolder = /usr/local/bin
```
Comment (AKalinich-Luxoft) [inline src/appMain/smartDeviceLink.ini:43]: @alexkutsan I see many paths changes in ini. Not sure if all of them are applicable/best for Linux environment.
Maybe it makes sense to create a separate ini file for a deb package and replace it when DEB-package build is requested?

File: src/appMain/log4cxx.properties [hunk 5]
```diff
@@ -45,7 +45,7 @@ log4j.appender.ProtocolFordHandlingLogFile.layout.ConversionPattern=%-5p [%d{dd
 
 # Root logging settings
 # Set ALL logging levels to main log file
-log4j.rootLogger=ALL, SmartDeviceLinkCoreLogFile
+log4j.rootLogger=ALL, Console, SmartDeviceLinkCoreLogFile
```
Comment (AKalinich-Luxoft) [inline src/appMain/log4cxx.properties:48]: @alexkutsan same suggestion as for ini file

File: src/appMain/main.cc [hunk 6]
```diff
@@ -133,7 +133,7 @@ int32_t main(int32_t argc, char** argv) {
 
   // --------------------------------------------------------------------------
   // Logger initialization
-  INIT_LOGGER("log4cxx.properties", profile_instance.logs_enabled());
+  INIT_LOGGER("/usr/local/bin/log4cxx.properties", profile_instance.logs_enabled());
```
Comment (AKalinich-Luxoft) [inline src/appMain/main.cc:136]: @alexkutsan maybe its possible to take this path from `profile_instance`?

File: CMakeLists.txt (hunk 1 above)
Comment (LuxoftAKutsan) [reply CMakeLists.txt:249]: New version of compiler on rasbian not compatible with logger and boost with warnings as errors= on

File: src/components/application_manager/src/plugin_manager/rpc_plugin_manager_impl.cc (hunk 2 above)
Comment (LuxoftAKutsan) [reply src/components/application_manager/src/plugin_manager/rpc_plugin_manager_impl.cc:40]: SDL try to load all files from specified dir. And consider that `so` is suitable is export symbols successful. So failed export is not an error, it is regular situation

File: src/components/security_manager/src/ssl_context_impl.cc (hunk 3 above)
Comment (LuxoftAKutsan) [reply src/components/security_manager/src/ssl_context_impl.cc:599]: Yep. nerw version of libssl is not compatible with previous code

File: src/appMain/smartDeviceLink.ini (hunk 4 above)
Comment (LuxoftAKutsan) [reply src/appMain/smartDeviceLink.ini:43]: There are several tricks in cmake to perform such staff. I will try to implement
